2023-10-17 19:50:19,374 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py changed, reloading.
2023-10-17 19:57:21,555 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py changed, reloading.
2023-10-17 19:58:16,662 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py changed, reloading.
2023-10-17 20:01:00,821 [ERROR]: Stripe invoice send operation failed: 'str' object has no attribute 'id'
2023-10-17 20:02:23,437 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py changed, reloading.
2023-10-17 20:03:12,214 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py changed, reloading.
2023-10-17 20:03:18,549 [ERROR]: Stripe invoice send operation failed: 'dict' object has no attribute 'id'
2023-10-17 20:08:41,372 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\Gallery_Project\env\app_Logic\stripe\quick_stripe.py changed, reloading.
2023-10-17 20:08:56,206 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py changed, reloading.
2023-10-17 20:09:15,066 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py changed, reloading.
2023-10-17 20:09:17,527 [ERROR]: Stripe invoice send operation failed: cannot access local variable 'payment_link' where it is not associated with a value
2023-10-17 20:09:50,733 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py changed, reloading.
2023-10-17 20:09:53,073 [ERROR]: Stripe invoice send operation failed: Request req_WUXWcLabQ5Hi9t: Invoice is already paid
2023-10-17 20:17:00,284 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py changed, reloading.
2023-10-17 20:17:03,511 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py changed, reloading.
2023-10-17 20:18:05,781 [ERROR]: Stripe invoice send operation failed: Unrecognized request URL (POST: /v1/invoices//send). If you are trying to list objects, remove the trailing slash. If you are trying to retrieve an object, make sure you passed a valid (non-empty) identifier in your code. Please see https://stripe.com/docs or we can help at https://support.stripe.com/.
2023-10-17 20:20:39,086 [ERROR]: Internal Server Error: /admin/auth/user/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1983, in changelist_view
    response = self.response_action(
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1593, in response_action
    response = func(self, request, queryset)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\actions.py", line 51, in delete_selected
    modeladmin.delete_queryset(request, queryset)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1237, in delete_queryset
    queryset.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\query.py", line 1148, in delete
    deleted, _rows_count = collector.delete()
                           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-17 20:22:45,937 [ERROR]: Internal Server Error: /admin/clients/client/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1983, in changelist_view
    response = self.response_action(
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1593, in response_action
    response = func(self, request, queryset)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\actions.py", line 51, in delete_selected
    modeladmin.delete_queryset(request, queryset)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1237, in delete_queryset
    queryset.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\query.py", line 1148, in delete
    deleted, _rows_count = collector.delete()
                           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-17 20:24:37,411 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\clients\models.py changed, reloading.
2023-10-17 20:24:56,035 [ERROR]: Internal Server Error: /admin/clients/client/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1983, in changelist_view
    response = self.response_action(
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1593, in response_action
    response = func(self, request, queryset)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\actions.py", line 51, in delete_selected
    modeladmin.delete_queryset(request, queryset)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1237, in delete_queryset
    queryset.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\query.py", line 1148, in delete
    deleted, _rows_count = collector.delete()
                           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-17 20:25:34,213 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\clients\models.py changed, reloading.
2023-10-17 20:25:38,622 [ERROR]: Internal Server Error: /admin/clients/client/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1983, in changelist_view
    response = self.response_action(
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1593, in response_action
    response = func(self, request, queryset)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\actions.py", line 51, in delete_selected
    modeladmin.delete_queryset(request, queryset)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1237, in delete_queryset
    queryset.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\query.py", line 1148, in delete
    deleted, _rows_count = collector.delete()
                           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-17 20:25:48,830 [ERROR]: Internal Server Error: /admin/clients/client/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1983, in changelist_view
    response = self.response_action(
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1593, in response_action
    response = func(self, request, queryset)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\actions.py", line 51, in delete_selected
    modeladmin.delete_queryset(request, queryset)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1237, in delete_queryset
    queryset.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\query.py", line 1148, in delete
    deleted, _rows_count = collector.delete()
                           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-17 20:26:21,276 [ERROR]: Internal Server Error: /admin/clients/client/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1983, in changelist_view
    response = self.response_action(
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1593, in response_action
    response = func(self, request, queryset)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\actions.py", line 51, in delete_selected
    modeladmin.delete_queryset(request, queryset)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1237, in delete_queryset
    queryset.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\query.py", line 1148, in delete
    deleted, _rows_count = collector.delete()
                           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-17 20:26:54,976 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\clients\models.py changed, reloading.
2023-10-17 20:27:18,362 [ERROR]: Internal Server Error: /admin/clients/client/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1983, in changelist_view
    response = self.response_action(
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1593, in response_action
    response = func(self, request, queryset)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\actions.py", line 51, in delete_selected
    modeladmin.delete_queryset(request, queryset)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1237, in delete_queryset
    queryset.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\query.py", line 1148, in delete
    deleted, _rows_count = collector.delete()
                           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-17 20:27:31,527 [ERROR]: Internal Server Error: /admin/clients/client/22/delete/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 2106, in delete_view
    return self._delete_view(request, object_id, extra_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 2142, in _delete_view
    self.delete_model(request, obj)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1233, in delete_model
    obj.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\base.py", line 1132, in delete
    return collector.delete()
           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-17 20:28:04,683 [ERROR]: Internal Server Error: /admin/clients/client/22/delete/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 2106, in delete_view
    return self._delete_view(request, object_id, extra_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 2142, in _delete_view
    self.delete_model(request, obj)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1233, in delete_model
    obj.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\base.py", line 1132, in delete
    return collector.delete()
           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-17 20:28:17,958 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\clients\models.py changed, reloading.
2023-10-17 20:29:11,163 [ERROR]: Internal Server Error: /clients/client/delete/22
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 278, in post
    return self.form_valid(form)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 283, in form_valid
    success_url = self.get_success_url()
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 240, in get_success_url
    raise ImproperlyConfigured("No URL to redirect to. Provide a success_url.")
django.core.exceptions.ImproperlyConfigured: No URL to redirect to. Provide a success_url.
2023-10-17 20:29:33,299 [ERROR]: Internal Server Error: /management/invoice/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py", line 85, in billing_panel
    'outstanding': bal_out,
                   ^^^^^^^
UnboundLocalError: cannot access local variable 'bal_out' where it is not associated with a value
2023-10-17 20:34:52,261 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\management\views.py changed, reloading.
2023-10-17 20:35:57,196 [WARNING]: Not Found: /favicon.ico
2023-10-18 02:39:38,547 [ERROR]: Internal Server Error: /clients/intake
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 184, in post
    return super().post(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 153, in post
    return self.form_valid(form)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\clients\views.py", line 105, in form_valid
    hex_key = hexer.hex_gen()
              ^^^^^^^^^^^^^^^
TypeError: Hexer.hex_gen() takes 0 positional arguments but 1 was given
2023-10-18 11:26:11,510 [WARNING]: Not Found: /favicon.ico
2023-10-18 11:26:11,520 [WARNING]: Not Found: /favicon.ico
2023-10-18 17:25:11,859 [WARNING]: Not Found: /favicon.ico
2023-10-18 17:26:34,568 [ERROR]: Internal Server Error: /clients/client/delete/22
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 278, in post
    return self.form_valid(form)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 283, in form_valid
    success_url = self.get_success_url()
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 240, in get_success_url
    raise ImproperlyConfigured("No URL to redirect to. Provide a success_url.")
django.core.exceptions.ImproperlyConfigured: No URL to redirect to. Provide a success_url.
2023-10-18 17:27:55,204 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\clients\views.py changed, reloading.
2023-10-18 17:28:00,843 [ERROR]: Internal Server Error: /clients/client/delete/22
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 278, in post
    return self.form_valid(form)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 284, in form_valid
    self.object.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\base.py", line 1132, in delete
    return collector.delete()
           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-18 17:29:55,042 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\gallery\models.py changed, reloading.
2023-10-18 17:29:57,680 [ERROR]: Internal Server Error: /clients/client/delete/22
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 278, in post
    return self.form_valid(form)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 284, in form_valid
    self.object.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\base.py", line 1132, in delete
    return collector.delete()
           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-18 17:30:31,552 [ERROR]: Internal Server Error: /clients/client/delete/22
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 278, in post
    return self.form_valid(form)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\generic\edit.py", line 284, in form_valid
    self.object.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\base.py", line 1132, in delete
    return collector.delete()
           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-18 22:40:17,340 [ERROR]: Internal Server Error: /clients/22/details
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\clients\views.py", line 85, in client_details
    user_info = User.objects.get(username=client_info.name)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\query.py", line 637, in get
    raise self.model.DoesNotExist(
django.contrib.auth.models.User.DoesNotExist: User matching query does not exist.
2023-10-18 22:40:50,739 [ERROR]: Internal Server Error: /admin/clients/client/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1983, in changelist_view
    response = self.response_action(
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1593, in response_action
    response = func(self, request, queryset)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\actions.py", line 51, in delete_selected
    modeladmin.delete_queryset(request, queryset)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1237, in delete_queryset
    queryset.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\query.py", line 1148, in delete
    deleted, _rows_count = collector.delete()
                           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-18 22:41:02,432 [ERROR]: Internal Server Error: /clients/22/details
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\clients\views.py", line 85, in client_details
    user_info = User.objects.get(username=client_info.name)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\query.py", line 637, in get
    raise self.model.DoesNotExist(
django.contrib.auth.models.User.DoesNotExist: User matching query does not exist.
2023-10-18 22:41:47,043 [ERROR]: Internal Server Error: /admin/clients/client/22/delete/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 2106, in delete_view
    return self._delete_view(request, object_id, extra_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 2142, in _delete_view
    self.delete_model(request, obj)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1233, in delete_model
    obj.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\base.py", line 1132, in delete
    return collector.delete()
           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-18 22:43:17,205 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\clients\models.py changed, reloading.
2023-10-18 22:43:46,307 [ERROR]: Internal Server Error: /admin/auth/user/
Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 688, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\views\decorators\cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\sites.py", line 242, in inner
    return view(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\utils\decorators.py", line 134, in _wrapper_view
    response = view_func(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1983, in changelist_view
    response = self.response_action(
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1593, in response_action
    response = func(self, request, queryset)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\actions.py", line 51, in delete_selected
    modeladmin.delete_queryset(request, queryset)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\contrib\admin\options.py", line 1237, in delete_queryset
    queryset.delete()
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\query.py", line 1148, in delete
    deleted, _rows_count = collector.delete()
                           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\deletion.py", line 506, in delete
    count = query.delete_batch(pk_list, self.using)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 42, in delete_batch
    num_deleted += self.do_query(
                   ^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\subqueries.py", line 20, in do_query
    cursor = self.get_compiler(using).execute_sql(CURSOR)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\models\sql\compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 102, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\django\db\backends\mysql\base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\cursors.py", line 330, in _query
    db.query(q)
  File "C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Photo_Gallery_project\Lib\site-packages\MySQLdb\connections.py", line 255, in query
    _mysql.connection.query(self, query)
django.db.utils.IntegrityError: (1217, 'Cannot delete or update a parent row: a foreign key constraint fails')
2023-10-18 22:45:26,539 [INFO]: C:\Users\joshu\Documents\Code Book\Websites\SoftSubversion.com\Gallery_Project\clients\models.py changed, reloading.
